### PR TITLE
Added tests for DateTime precision.

### PR DESCRIPTION
**Pull request by tvrprasad**
This adds tests for the PR https://github.com/tediousjs/tedious/pull/517.

### CODE REVIEW

**Comment by tvrprasad** [comment]
This is odd. Tests pass for me for all versions of TDS, both against 2016 and 2012 versions of SQL sever. I'll experiment with some tweaks to the PR. I'll clean up before merge.

File: test/integration/parameterised-statements-test.coffee [hunk 1]
```diff
@@ -286,6 +320,40 @@ exports.outputDateTime = (test) ->
...
 
+# The tests below validates DateTime precision on the output side, as described in
+#  the section "Rounding of datetime Fractional Second Precision" from
+# https://msdn.microsoft.com/en-us/library/ms187819.aspx
+
+exports.outputDatePrecision_0 = (test) ->
+  execSqlOutput(test, TYPES.DateTime, new Date('January 1, 1998 23:59:59.990'))
+
+exports.outputDatePrecision_1 = (test) ->
+  execSqlOutput(test, TYPES.DateTime, new Date('January 1, 1998 23:59:59.990'))
```
Comment (tvrprasad) [inline test/integration/parameterised-statements-test.coffee]: Got these tests half-way done :-(
Will take care of this.

**Comment by tvrprasad** [comment]
The two failing, based on the debug logging added, are dateTimePrecision_9 and outputDatePrecision_9. They both send new Date('January 1, 1998 23:59:59.999') to SQL Server as DateTime. I've tried this on two different versions of SQL server - one local and one Azure hosted. Works on both. This makes me think this is a bug in the specific SQL server version being used in AppVeyor tests. I'm updating the PR to add a test to just print out the SQL Server version to see what it is.

SQL Server versions I tried:

Microsoft SQL Server 2016 (RTM-GDR) (KB3194716) - 13.0.1722.0 (X64)   Sep 26 2016 13:17:23   Copyright (c) Microsoft Corporation  Enterprise Edition: Core-based Licensing (64-bit) on Windows 10 Enterprise 6.3 <X64> (Build 14393: ) 

Microsoft SQL Azure (RTM) - 12.0.2000.8   Feb  8 2017 04:15:27   Copyright (C) 2016 Microsoft Corporation. All rights reserved.

**Comment by tvrprasad** [comment]
The version of SQL server on AppVeyor is older by about 5 months.

Microsoft SQL Server 2016 (RTM) - 13.0.1601.5 (X64) 
	Apr 29 2016 23:23:58 
	Copyright (c) Microsoft Corporation
	Developer Edition (64-bit) on Windows Server 2012 R2 Datacenter 6.3 <X64> (Build 9600: ) (Hypervisor)

**Comment by arthurschreiber** [comment]
@tvrprasad That seems like a really weird edge case. 😞

I think it would definitely be ok to change these test cases to not trigger the failure as long as the coverage does not increase.

**Comment by tvrprasad** [comment]
@arthurschreiber That sounds good. I'll try a couple of other things and push this out (after cleanup) without the failing test cases. Perhaps we can look into upgrading the SQL Server version for our tests?

**Comment by tvrprasad** [comment]
This one fails:
'January 1, 1998 23:59:59.999'

But these succeed:
'January 1, 1998 23:59:58.999'
'January 1, 1998 23:58:59.999'
'January 1, 1998 22:59:59.999'

A very, very corner case :-)
